AnimationSystem/AnimationSystem.h

```diff
@@ -0,0 +1,133 @@
+#ifndef _ANIMATIONSYSTEM_H
+#define _ANIMATIONSYSTEM_H
+#include "physics.h"
+#include <iostream>
+
+namespace AnimationSystem {
+  struct Interpolant;
+  struct Animation;
+  struct AnimationMapping;
+  class Avatar;
+  class AnimationNode;
+  class AnimationMixer;
+
+  struct Interpolant {
+    unsigned int numParameterPositions;
+    float *parameterPositions;
+    float *resultBuffer;
+    unsigned int numSampleValues;
+    float *sampleValues;
+    unsigned int valueSize;
+  };
+  struct Animation {
+    float duration;
+    std::vector<Interpolant> interpolants;
+    unsigned int currentInterpolantIndex = 0;
+    unsigned int index;
+    std::string name;
+  };
+  struct AnimationMapping { // spec
+    float *dst =  (float *)malloc((4) * sizeof(float));
```

Comment:
What's this for? Not generally good to have this random malloc()...

Reply:
It's used for keep animation result data like what js version's `animationMapping.dst` do.
Because of `animationMappings` only need init once globally, and will return these `dst` values back to js, so I allocated them.
https://github.com/webaverse/app-wasm/pull/75/files#diff-71459e596906bb9318622660fe80e75db679de846febc40fc79f6f81e48a7ed3R1069
Let me try if there's nicer way to do it.

Reply:
For performance reason, I directly return `dst`s' pointers to js side.
https://github.com/webaverse/app-wasm/blob/808c4102b91357d9f31cdc793f2f5bf9236eb6eb/AnimationSystem/AnimationSystem.cc#L1234-L1237
So I think have to use `heap` memory instead of `stack` memory.
But I used `new float[size]` instead of `(float *)malloc((size) * sizeof(float))` for clearer codes.
https://github.com/webaverse/app-wasm/pull/75/commits/808c4102b91357d9f31cdc793f2f5bf9236eb6eb
Is it ok?

Reply:
No, I still have no idea what this is about. There should never be a reason to initialize struct members to the heap like this.

If you need a global, use a global. Otherwise use an inline member like an array.

Reply:
From your other code I'm assuming you intended to use an inline array. Maybe you are confused about how to read it in JS?

Reply:
Used inline array and global singleton instead of initialize struct members to the heap.
https://github.com/webaverse/app-wasm/compare/b667920cc2bbda7da0c901cf2236cdf2c3c99e8d...b56cd596f08bc8147647384576cab9b44352f961

Reply:
I don't get it. Why does it need `malloc`?